service/prometheus/service.go

```diff
@@ -134,6 +141,14 @@ func (s *Service) isReloadRequired() (bool, error) {
 		return false, nil
 	}
 
+	if s.isReloadRequested {
```

Comment:
We should use the public interface here since this will ensure the synchronization.

Reply:
Again, we do this all sequentially, I'm not too worried with that.